### PR TITLE
FDG-1046 Set Input guardrail by Checking Size

### PR DESCRIPTION
**Pull request by bhatiakartik**

### CODE REVIEW

File: flipkart-guardrails/src/main/java/com/flipkart/fdp/AdjustJobConfigurationImpl.java [hunk 1]
```diff
@@ -125,6 +129,11 @@ public void adjustJobConfiguration(Configuration jobConf) {
     }
 
     ensureAllPropertiesAreAvailable(processDataProperties);
+    try{
+      checkInputFilesSize(jobConf);
+    }catch (IOException io){
+      new RuntimeException("IO exception occured while checking input paths");
```
Comment (himanshuvirmani) [inline flipkart-guardrails/src/main/java/com/flipkart/fdp/AdjustJobConfigurationImpl.java:135]: Please mention here what was the size calculated and what was the upperbound configured

File: flipkart-guardrails/src/main/java/com/flipkart/fdp/AdjustJobConfigurationImpl.java [hunk 2]
```diff
@@ -197,4 +206,30 @@ public void adjustJobConfiguration(Configuration jobConf) {
     }
     jobConf.set(Constants.PASS_THROUGH, "false");
   }
+
+  private void checkInputFilesSize(Configuration jobConf) throws IOException {
+    Path [] inputPaths = getInputPaths(jobConf);
+    FileSystem fs = FileSystem.get(jobConf);
+    Long totalInputSizeInBytes = 0L;
+    Long readThreshold = jobConf.getLong(Constants.READ_THRESHOLD, -1l);
```
Comment (himanshuvirmani) [inline flipkart-guardrails/src/main/java/com/flipkart/fdp/AdjustJobConfigurationImpl.java]: readTheshold can be declared final here.

File: flipkart-guardrails/src/main/java/com/flipkart/fdp/AdjustJobConfigurationImpl.java [hunk 3]
```diff
@@ -197,4 +206,30 @@ public void adjustJobConfiguration(Configuration jobConf) {
...
+      totalInputSizeInBytes += fs.getContentSummary(inputPath).getLength();
+      if(totalInputSizeInBytes > readThreshold){
+        throw new RuntimeException("Input size cannot exceed " + readThreshold);
+      }
+    }
+  }
+
+  private Path[] getInputPaths(Configuration jobConf) {
+    String dirs = jobConf.get(INPUT_DIR, "");
+    String [] list = StringUtils.split(dirs);
```
Comment (himanshuvirmani) [inline flipkart-guardrails/src/main/java/com/flipkart/fdp/AdjustJobConfigurationImpl.java]: Use trim before split

Comment (himanshuvirmani) [inline flipkart-guardrails/src/main/java/com/flipkart/fdp/AdjustJobConfigurationImpl.java]: How is this making sure to include recursive files in dirs..

File: flipkart-guardrails/src/main/java/com/flipkart/fdp/AdjustJobConfigurationImpl.java (hunk 1 above)
Comment (bhatiakartik) [reply flipkart-guardrails/src/main/java/com/flipkart/fdp/AdjustJobConfigurationImpl.java:135]: This is the IO exception not exception of FileSystem.getContentSummary method. For input file size exception it has been mentioned.

File: flipkart-guardrails/src/main/java/com/flipkart/fdp/AdjustJobConfigurationImpl.java (hunk 2 above)
Comment (bhatiakartik) [reply flipkart-guardrails/src/main/java/com/flipkart/fdp/AdjustJobConfigurationImpl.java]: done

File: flipkart-guardrails/src/main/java/com/flipkart/fdp/AdjustJobConfigurationImpl.java (hunk 3 above)
Comment (bhatiakartik) [reply flipkart-guardrails/src/main/java/com/flipkart/fdp/AdjustJobConfigurationImpl.java]: done

**Review by himanshuvirmani — APPROVED** [review]

**Pull request merged**